lib/DoctrineCodingStandard/Sniffs/Classes/ExceptionClassNamingSniff.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DoctrineCodingStandard\Sniffs\Classes;
+
+use DoctrineCodingStandard\Helpers\UseStatementHelper;
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use const T_ABSTRACT;
+use const T_CLASS;
+use const T_FINAL;
+use function count;
+use function preg_grep;
+use function preg_match;
+
+final class ExceptionClassNamingSniff implements Sniff
+{
+    private const CODE_NOT_AN_EXCEPTION_CLASS = 'NotAnExceptionClass';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function register() : array
+    {
+        return [T_CLASS];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(File $phpcsFile, $stackPtr) : void
+    {
+        $isAbstract              = $phpcsFile->findFirstOnLine([T_ABSTRACT], $stackPtr) !== false;
+        $isFinal                 = $phpcsFile->findFirstOnLine([T_FINAL], $stackPtr) !== false;
```

Comment:
Since class can't be both abstract and final, you this should preferably use `SlevomatCodingStandard\Helpers\TokenHelper::findPreviousEffective($phpcsFile, $pointer)` and check for `T_FINAL`/`T_ABSTRACT` (using this instead of `findFirstOnLine()` is also safer).